proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+To better understand the issue at hand, please take note of following example:
+Let us assume that a mobile application with a newer version of the RAPI attempts to connect to an SDL Core instance which is using an older RAPI version. In this example, the SDL Core RAPI version is set to 4.3 whereas the mobile application RAPI version is set to 6.0. Let us also assume that the mobile application has an AppHMIType of DEFAULT and REMOTE_CONTROL set.  The DEFAULT AppHMIType has been present in the RAPI since its inception, however REMOTE_CONTROL was introduced in version 6.0 of the RAPI.  
+
+In the case above, the mobile application will try to register itself by sending a RegisterAppInterface (RAI) request to SDL Core with an array of AppHMITypes : appHMIType[DEFAULT, REMOTE_CONTROL]. SDL Core will next try to validate the parameters in the RAI request, it will immediately find an unknown enum value of REMOTE_CONTROL. A failure response will then be sent to the mobile application with the INVALID_DATA result code (this is because the REMOTE_CONTROL AppHMIType has only been available since version RAPI 4.5, however we are communicating with a version of SDL Core that only supports RAPI 4.3). As a result of the scenario above, the mobile application fails to register with SDL Core providing an unacceptable user experience -- in addition the mobile application cannot conclusively determine the cause of failure since INVALID_DATA can be returned for any number of reasons.  Ideally the mobile app would be registered successfully with a warning provided in the result code stating some AppHMITypes could not be processed.
+
+To fix the case stated above as well as similar cases originating from the same cause, the validation logic inside of SDL core should be updated to cutoff / ignore unknown enums present in RPC requests.  These failure cases should instead be replaced with SUCCESS cases with warnings result codes.  
+To ensure backward compatibility with earlier versions of SDL core, mobile libraries should be updated to process, filter and format enums inside of RPC requests when a difference in RAPI sets have been detected.
+
+  ## Proposed solution
+
+  ### SDL Core changes
+
+  When SDL receives a request from Mobile Application it starts validation of all incoming parameters.
+  If a parameter includes unknown enum value, SDL core has to cut off such value from the parameter.
+  This means the following cases are possible:
+  1. Type of parameter is non-array enum (see example 1)
+      SDL core has to remove this parameter from the request since the value after cutting off becomes empty.
```

Comment:
```suggestion
      SDL Core has to remove this parameter from the request since the value after cutting off becomes empty.
```
Please also note that this is appearing on the same line as "1. Type of parameter..."  If you want these to be on different lines, you'll need to add a line break.